File: Ruby/uri-1048.rb
# frozen_string_literal: true

# URI - 1048: https://www.urionlinejudge.com.br/judge/pt/problems/view/1048

def readjust_salary(salary)
  if salary <= 400.00
    calculate_new_salary(salary, 0.15)
  elsif salary <= 800.00
    calculate_new_salary(salary, 0.12)
  elsif salary <= 1200.00
    calculate_new_salary(salary, 0.10)
  elsif salary <= 2000.00
    calculate_new_salary(salary, 0.07)
  else
    calculate_new_salary(salary, 0.04)
  end
end

def calculate_new_salary(salary, readjustment_percentage)
  gain = salary * readjustment_percentage
  {
    new_salary: (salary + gain),
    gain_adjustment: gain,
    percentage_gain: "#{(readjustment_percentage * 100).to_i} %"
  }
end

salary = gets.to_f
new_salary_data = readjust_salary(salary)
puts(format('Novo salario: %<new_salary>.2f', new_salary: new_salary_data[:new_salary]))
puts(format('Reajuste ganho: %<gain_adjustment>.2f', gain_adjustment: new_salary_data[:gain_adjustment]))
puts "Em percentual: #{new_salary_data[:percentage_gain]}"

File: URI/JavaScript/uri-1048.js
// URI - 1048: https://www.urionlinejudge.com.br/judge/pt/problems/view/1048

const input = require('fs').readFileSync('/dev/stdin', 'utf-8');
const salary = parseFloat(input);

const newSalary = calculateNewSalary(salary);
console.log('Novo salario: ' + newSalary.newSalary);
console.log('Reajuste ganho: ' + newSalary.readjustment);
console.log('Em percentual: ' + newSalary.percentage);

function calculateNewSalary(currentSalary) {
  const percentage = getPercentageForSalary(currentSalary);
  const newSalary = (currentSalary + (currentSalary * (percentage / 100))).toFixed(2);
  const readjustment = (newSalary - currentSalary).toFixed(2);
  return { newSalary, readjustment, percentage: percentage.toString() + ' %' };
}

function getPercentageForSalary(salary) {
  if (salary > 0 && salary <= 400.00)
    return 15;
  else if (salary <= 800.00)
    return 12;
  else if (salary <= 1200.00)
    return 10;
  else if (salary <= 2000.00)
    return 7;
  else
    return 4;
}
